### PR TITLE
Convert openssl resources to unified_mode

### PR DESCRIPTION
**Pull request by lamont-granquist**
Mostly does what the title says, there wasn't much that depended on
compile/converge ordering and mostly this might fix a bug or two
in intention.

Renamed a method that had a misspelling.

Used tap in two places.

Removed the mutation of the new_resource in two places which is
the most disruptive part of this change.

### CODE REVIEW

File: lib/chef/resource/openssl_x509_certificate.rb [hunk 1]
```diff
@@ -185,24 +187,27 @@ class OpensslX509Certificate < Chef::Resource
...
+          @key_file ||=
+            begin
+              if new_resource.key_file
+                new_resource.key_file
+              else
+                path, file = ::File.split(new_resource.path)
+                filename = ::File.basename(file, ::File.extname(file))
+                path + "/" + filename + ".key"
+              end
+            end
```
Comment (phiggins) [inline lib/chef/resource/openssl_x509_certificate.rb:200]:
```suggestion
            if new_resource.key_file
              new_resource.key_file
            else
              path, file = ::File.split(new_resource.path)
              filename = ::File.basename(file, ::File.extname(file))
              path + "/" + filename + ".key"
            end
```

The `begin...end` is superfluous here.

File: lib/chef/resource/openssl_x509_request.rb [hunk 2]
```diff
@@ -145,36 +147,39 @@ class OpensslX509Request < Chef::Resource
...
+          @key_file ||=
+            begin
+              if new_resource.key_file
+                new_resource.key_file
+              else
+                path, file = ::File.split(new_resource.path)
+                filename = ::File.basename(file, ::File.extname(file))
+                path + "/" + filename + ".key"
+              end
+            end
```
Comment (phiggins) [inline lib/chef/resource/openssl_x509_request.rb:160]:
```suggestion
            if new_resource.key_file
              new_resource.key_file
            else
              path, file = ::File.split(new_resource.path)
              filename = ::File.basename(file, ::File.extname(file))
              path + "/" + filename + ".key"
            end
```

`begin...end` is superfluous here too.

File: lib/chef/resource/openssl_x509_request.rb [hunk 3]
```diff
@@ -145,36 +147,39 @@ class OpensslX509Request < Chef::Resource
...
-          csr_subject.add_entry("emailAddress", new_resource.email) unless new_resource.email.nil?
-          csr_subject
+          OpenSSL::X509::Name.new.tap do |csr_subject|
+            csr_subject.add_entry("C", new_resource.country) unless new_resource.country.nil?
+            csr_subject.add_entry("ST", new_resource.state) unless new_resource.state.nil?
+            csr_subject.add_entry("L", new_resource.city) unless new_resource.city.nil?
+            csr_subject.add_entry("O", new_resource.org) unless new_resource.org.nil?
+            csr_subject.add_entry("OU", new_resource.org_unit) unless new_resource.org_unit.nil?
+            csr_subject.add_entry("CN", new_resource.common_name)
+            csr_subject.add_entry("emailcsr_subject.address", new_resource.email) unless new_resource.email.nil?
```
Comment (tas50) [inline lib/chef/resource/openssl_x509_request.rb:181]: ?

File: lib/chef/resource/openssl_x509_certificate.rb [hunk 4]
```diff
@@ -214,15 +219,15 @@ def request
...
-          subject.add_entry("emailAddress", new_resource.email) unless new_resource.email.nil?
-          subject
+          OpenSSL::X509::Name.new.tap do |csr_subject|
+            csr_subject.add_entry("C", new_resource.country) unless new_resource.country.nil?
+            csr_subject.add_entry("ST", new_resource.state) unless new_resource.state.nil?
+            csr_subject.add_entry("L", new_resource.city) unless new_resource.city.nil?
+            csr_subject.add_entry("O", new_resource.org) unless new_resource.org.nil?
+            csr_subject.add_entry("OU", new_resource.org_unit) unless new_resource.org_unit.nil?
+            csr_subject.add_entry("CN", new_resource.common_name)
+            csr_subject.add_entry("emailcsr_subject.address", new_resource.email) unless new_resource.email.nil?
```
Comment (phiggins) [inline lib/chef/resource/openssl_x509_certificate.rb:229]: Was this intentional? I can't find any reference to `emailcsr_subject` on the web.

Comment (lamont-granquist) [reply lib/chef/resource/openssl_x509_certificate.rb:227]: search and replace went crazy

**Pull request merged**